eventsources/sources/bitbucketserver/start.go

```diff
@@ -709,3 +670,165 @@ func (c *customBitbucketClient) GetCommitPullRequests(project, repository, commi
 
 	return pullRequests, nil
 }
+
+// webhookHandler processes Bitbucket webhook events specifically looking for "repo:refs_changed" events.
+// It first checks if the incoming webhook is configured to be handled (i.e., it's a "repo:refs_changed" event).
+// If not, it logs the discrepancy and returns the raw event.
+//
+// For valid "repo:refs_changed" events, it:
+//   - Unmarshals the JSON body into a refsChangedWebhookEvent struct.
+//   - Optionally filters out changes associated with open pull requests if SkipBranchRefsChangedOnOpenPR is enabled.
+//   - Depending on the OneEventPerChange setting, it either batches all changes into one event or creates individual events for each change.
+//
+// It returns a slice of byte slices containing the processed or original webhook events, and any error encountered during processing.
+func webhookHandler(request *http.Request, logger *zap.SugaredLogger, body []byte, bitbucketserverEventSource *v1alpha1.BitbucketServerEventSource, router *Router) ([][]byte, error) {
+	if request.Header.Get("X-Event-Key") != "repo:refs_changed" {
+		// Don't continue if this is not a repo:refs_changed webhook event.
```

Comment:
Is this the same as the previous logic?

Reply:
Pretty much the same except I learned that Bitbucket sends the event type/key in a HTTP header named `X-Event-Key`. Which means we don't have to decode every event now. Making the code easier to read and little more performant.

<img width="1000" alt="Screenshot 2024-05-29 at 10 21 33 PM" src="https://github.com/argoproj/argo-events/assets/6209771/14904d28-f883-4497-951b-67422a1677c1">
<img width="985" alt="Screenshot 2024-05-29 at 10 23 23 PM" src="https://github.com/argoproj/argo-events/assets/6209771/45a0d0af-08fe-4a3f-9933-9e0c4448dc53">

Reply:
@whynowy does that explanation help?